refarch-integrations/refarch-s3-integration/refarch-s3-integration-service/src/main/java/de/muenchen/refarch/s3/integration/configuration/nfcconverter/NfcReader.java

```diff
@@ -0,0 +1,105 @@
+package de.muenchen.refarch.s3.integration.configuration.nfcconverter;
+
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.io.Reader;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * <p>
+ * Wrapper for readers that performs an NFC conversion.
+ * </p>
+ *
+ * <p>
+ * <strong>Warning:</strong>
+ * <ul>
+ * <li>With Java readers and writers, an NFC conversion can be carried out safely as characters are
+ * processed there.</li>
+ * <li>Before reading the first character, this reader reads the complete text of the wrapped reader
+ * into an internal buffer and
+ * then carries out the NFC normalisation. The reason is that NFC conversion cannot be performed on
+ * the basis of individual characters.
+ * This can lead to increased latency.</li>
+ * </ul>
+ * </p>
+ */
```

Comment:
Why a JavaDoc comment using HTML, we don't have this anywhere else, would rewrite. Please check in other files as well.

Reply:
I shortened the html comments a bit but there is not other way for lists and think that's better suitable here.